packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -402,31 +407,90 @@ const SideNav = ({
     [isCompact],
   );
 
-  const processedSideNavItems = useMemo(
-    () =>
-      sideNavItems.map((item) => ({
-        ...item,
-        children: item.children?.map((childProps) => {
-          return (
+  const setSelectedInChildItems = useCallback(
+    ({
+      nestedNavItems,
+      selectedItemId,
+    }: {
+      nestedNavItems: Array<Omit<SideNavItem, "startIcon" | "endIcon">>;
+      selectedItemId: string;
+    }) => {
+      const childItems = nestedNavItems.map((childItem) => {
+        if (childItem.isSelected) {
+          childItem.isSelected = false;
+        }
+        if (childItem.id === selectedItemId) {
+          childItem.isSelected = true;
+        }
+        if (childItem.nestedNavItems) {
+          setSelectedInChildItems({
+            nestedNavItems: childItem.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return childItem;
+      });
+      return childItems;
+    },
+    [],
+  );
```

Comment:
We can avoid these mutations.

In general, it's best practice to avoid all `value = newValue` situations. Mutations have too many situations where something could go wrong, and they make maintenance harder even if it's a local variable.

But the real issue here is that it's not a local variable, we're mutating the object we were passed into this function inside the `.map` and changing those values in realtime. That can cause may unexpected issues.

I refactored your function to be an implicit return for everything:

```suggestion
const setSelectedInChildItems = useCallback(
  ({
    nestedNavItems,
    selectedItemId,
  }: {
    nestedNavItems: Array<Omit<SideNavItem, "startIcon" | "endIcon">>;
    selectedItemId: string;
  }) => (
    nestedNavItems.map((childItem) => ({
      ...childItem,
      isSelected: (
        childItem.id === selectedItemId
      ),
      nestedNavItems: (
        childItem.nestedNavItems
        ? (
          setSelectedInChildItems({
            nestedNavItems: childItem.nestedNavItems,
            selectedItemId,
          })
        )
        : undefined
      ),
    });
  ),
  [],
);
```